scripts/test_lib.sh

```diff
@@ -452,3 +452,36 @@ function git_assert_branch_in_sync {
     log_warning "Cannot verify consistency with the origin, as git is on detached branch."
   fi
 }
+
+# Generic function to execute specific test functions
+function test_executor {
+  local test_function="${1}"
+  shift 1
+  log_callout -e "\\n'${test_function}' started at $(date)"
+  if "${test_function}" "$@" ; then
+    log_success "'${test_function}' PASSED and completed at $(date)"
+    return 0
+  else
+    log_error "FAIL: '${test_function}' FAILED at $(date)"
+    if [ "$KEEP_GOING_SUITE" = true ]; then
+      return 2
+    else
+      exit 255
+    fi
+  fi
+}
+
+# execute integration test and e2e tests
+function integration_e2e_pass {
+  run_pass "integration" "${@}"
+  test_executor "e2e_test" "${@}"
+}
```

Comment:
This `integration_e2e_pass` is identical to the `integration_e2e_pass` in test.sh. Why there are two identical functions?